hypervisor/virt/rhevm/rhevmcli.py

```diff
@@ -88,3 +88,98 @@ def info(self):
             hosts = list()
         logger.info(f"Get RHEVM Host: {hosts}")
         return hosts
+
+    def guest_search(self, guest_name, host_ip, host_user, host_pwd):
+        """
+        Search the specific guest, return the expected attributes
+        :param guest_name: name for the specific guest
+        :param host_ip: the ip for the RHEVM host
+        :param host_user: the ssh username for the RHEVM host
+        :param host_pwd: the ssh password for the RHEVM host
+        :return: guest attributes, exclude guest_name, guest_ip, guest_uuid ...
+        """
+        host_id = self.get_rhevm_info("vm", guest_name, 'host-id')
+        cluster_id = self.get_rhevm_info("vm", guest_name, 'cluster-id')
+        guest_msgs = {
+            'guest_name': guest_name,
+            'guest_ip': self.get_guest_ip_by_mac(guest_name, host_ip, host_user, host_pwd),
+            'guest_uuid': self.get_rhevm_info("vm", guest_name, 'id'),
+            'guest_state': self.get_rhevm_info("vm", guest_name, 'status-state'),
+            'vdsm_uuid': host_id,
+            'vdsm_hwuuid': self.get_rhevm_info("host", host_id, 'hardware_information-uuid'),
+            'vdsm_hostname': self.get_rhevm_info("host", host_id, 'name'),
+            'vdsm_version': self.get_rhevm_info("host", host_id, 'version-full_version'),
+            'vdsm_cpu': self.get_rhevm_info("host", host_id, 'cpu-topology-sockets'),
+            'vdsm_cluster': self.get_rhevm_info("cluster", cluster_id, 'name'),
+        }
+        return guest_msgs
+
+    def get_rhevm_info(self, object_type, object_id, value):
+        """
+        Get the info from RHEVM
+        :param object_type: The type of object to retrieve, such as vm, host, cluster and so on.
+        :param object_id:  <id|name> The object identifier or the object identifier
+        :param value: the value you want to get from the result
+        :return: the value you want to get
+        """
+        cmd = f"ovirt-shell -c -E 'show {object_type} {object_id}' |grep '^{value}'"
+        ret, output = self.ssh.runcmd(cmd)
+        if not ret and value in output:
+            result = output.strip().split(':')[1].strip()
+            logger.info(f"Succeeded to get rhevm {object_type} ({object_id}) {value}: {result}")
+            return result
+        else:
+            logger.info(f"Failed to get rhevm {object_type} ({object_id}) {value}")
```

Comment:
Is it better to return a null value when failed to get the info? Or it returns None, right?